sql/sql_class.cc

```diff
@@ -2095,6 +2096,8 @@ int THD::killed_errno()
     DBUG_RETURN(ER_QUERY_INTERRUPTED);
   case KILL_TIMEOUT:
   case KILL_TIMEOUT_HARD:
+    if (slave_thread)
+      DBUG_RETURN(ER_SLAVE_STATEMENT_TIMEOUT);
     DBUG_RETURN(ER_STATEMENT_TIMEOUT);
```

Comment:
Indeed. The returned value can be constructed by the replication caller which is obviously better.

Reply:
There's a complication in the STATEMENT binlog format.
It is that in this case, unlike ROW format, `THD::main_da::m_sql_error` is set to `ER_STATEMENT_TIMEOUT` (that is to the general server error) deeply in the query exec stack. In ROW format I can easily overwrite it because and its stack the killed error is yet to be placed into `main_da`.

However, maybe we can stay with the general error reported on the slave as well? That would look like this
```
                Last_SQL_Error: Could not execute Write_rows_v1 event on table test.t1; Query execution was interrupted (max_statement_time exceeded), Error_code: 1969; the event's master log master-bin.000001, end_log_pos 925
```
Notice `max_statement_time exceeded` does not say it as perfect as `slave_max_statement_time`
would do. Anyway, I'd for this one.

Reply:
Sergei replies 
```
using ER_STATEMENT_TIMEOUT is kind of sloppy and might confuse users, better not to do it
``` which concludes this conversation.